tests/test_routes/test_flights_route.py

```diff
@@ -0,0 +1,372 @@
+        'rocket': stub_rocket,
+        'rail_length': 1,
+        'time_overshoot': True,
+        'terminate_on_apogee': True,
+        'equations_of_motion': 'STANDARD',
+    }
+    return flight
+
+
+@pytest.fixture
+def stub_flight_summary():
+    flight_summary = FlightSummary()
+    flight_summary_json = flight_summary.model_dump_json()
+    return json.loads(flight_summary_json)
+
+
+def test_create_flight(stub_flight):
+    with patch.object(
+        FlightController,
+        'create_flight',
+        return_value=FlightCreated(flight_id='123'),
+    ) as mock_create_flight:
+        with patch.object(
+            Motor, 'set_motor_kind', side_effect=None
+        ) as mock_set_motor_kind:
+            response = client.post(
+                '/flights/', json=stub_flight, params={'motor_kind': 'HYBRID'}
+            )
+            assert response.status_code == 200
+            assert response.json() == {
+                'flight_id': '123',
+                'message': 'Flight successfully created',
+            }
+            mock_set_motor_kind.assert_called_once_with(MotorKinds.HYBRID)
+            mock_create_flight.assert_called_once_with(Flight(**stub_flight))
+
+
+def test_create_flight_optional_params(stub_flight):
+    stub_flight.update(
+        {
+            'inclination': 0,
+            'heading': 0,
+            'max_time': 1,
+            'max_time_step': 1.0,
+            'min_time_step': 1,
+            'rtol': 1.0,
+            'atol': 1.0,
+            'verbose': True,
+        }
+    )
+    with patch.object(
+        FlightController,
+        'create_flight',
+        return_value=FlightCreated(flight_id='123'),
+    ) as mock_create_flight:
+        with patch.object(
+            Motor, 'set_motor_kind', side_effect=None
+        ) as mock_set_motor_kind:
+            response = client.post(
+                '/flights/', json=stub_flight, params={'motor_kind': 'HYBRID'}
+            )
+            assert response.status_code == 200
+            assert response.json() == {
+                'flight_id': '123',
+                'message': 'Flight successfully created',
+            }
+            mock_set_motor_kind.assert_called_once_with(MotorKinds.HYBRID)
+            mock_create_flight.assert_called_once_with(Flight(**stub_flight))
+
+
+def test_create_flight_invalid_input():
+    response = client.post(
+        '/flights/', json={'environment': 'foo', 'rocket': 'bar'}
+    )
+    assert response.status_code == 422
+
+
+def test_create_flight_server_error(stub_flight):
+    with patch.object(
+        FlightController,
+        'create_flight',
+        side_effect=HTTPException(
+            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR
+        ),
+    ):
+        response = client.post(
+            '/flights/', json=stub_flight, params={'motor_kind': 'HYBRID'}
+        )
+        assert response.status_code == 500
+        assert response.json() == {'detail': 'Internal Server Error'}
```

Comment:
What is the rationale for the scope of what is being mocked here? Correct-me if I understood it wrong, but the way I see this test:

1. The `create_flight` method is being mocked and its expected return is given and asserted;
2. The whole `POST` method in the `FlightController` is the `create_flight` and it is mocked on its entirety.

If those two things are correct, no code from the actual `FlightController` is being tested here, I believe, (except only by Pydantic and FastAPI inner workings). Shouldn't the only mocked dependency be the `Repository` of the `Controller`, so that actual code from it is used? 

The same comment applies to the other tests.

Reply:
Your observation is right, the reason is that we want to make the unit tests as modular as possible.

This whole series of PR's are all to test only (and exclusively) the project `routes`. Eventually we will have tests specific for the `controllers`, and ideally in the end we will have integration tests that validate everything altogether without mocked interfaces.

This is designed like that to ensure that if something breaks the issue can be quickly detected in isolation from the rest of the working pieces, which facilitates maintenance .

Reply:
Understood, being a route test makes sense, perhaps a bit too verbose, but it is better to have more than too few.